### PR TITLE
refactored signal's delight handler to deduplicate globally and track…

### PR DESCRIPTION
**Pull request by stephenlb**
… in librato.  This should allow us to track duplicates in librato, and also still process things like `import.js` even if it is a duplicate.  which is good.  👍

### CODE REVIEW

**Comment by stephenlb** [comment]
@raimondi1337 wdyt?

File: handlers/delights.js [hunk 1]
```diff
@@ -22,17 +22,17 @@ export default request => {
...
     // TODO special track signal 'reaction' for extra metrics
-    // TODO track sent delights
     // TODO send to SQL DB
 
-    // TODO Augment/Extend 'SYNC' User Profile KV Entry
+    // TODO Augment/Extend the User Profile in the KV Entry
     //      with all new keys supplied in the signal data
     //      so we have a progressively built profile.
-    
+    //      Right now it's just overriding the existing cutomer.
```
Comment (raimondi1337) [inline handlers/delights.js]: typo

File: handlers/delights.js [hunk 2]
```diff
@@ -22,17 +22,17 @@ export default request => {
...
+
+    // When processing a Non-delight
+    // such as running ./signals/import.js then
+    // we don't need the lookup a customer record
```
Comment (raimondi1337) [inline handlers/delights.js]: typo

File: signals/signup.js [hunk 3]
```diff
@@ -1,86 +1,78 @@
...
+    `Hi ${name || 'there'}! ` +
+    `I am Neumann, an Artificial Intelligence alive in PubNub BLOCKS. ` +
+    `Welcome to the Programmable Network.` +
+    `<p><p>` +
+    `${usecase ? 'I can help you with realtime ' + usecase + ' and more.' : ''} ` +
+    //`${(city || company && title) ? 'I know a little about you.' : ''} ` +
+    //`${city    ? 'I see you are in ' + city + '.' : ''} ` +
+    //`${title   ? 'You are the ' + title + (company?'at '+company:'') + '.' : ''} ` +
+    //`${company ? 'What is new at ' + company + '?' : ''} ` +
+    `${(city||'').indexOf('San') >= 0 ?
```
Comment (raimondi1337) [inline signals/signup.js:60]: This will hit developers in San Diego, San Jose etc too I think

**Review by raimondi1337 — COMMENTED** [review]
Dedupe code will also need to be gutted from Techcrunch and Producthunt, I can do that later orit can go in here doesn't matter

File: signals/signup.js (hunk 3 above)
Comment (stephenlb) [reply signals/signup.js:60]: yup!

**Comment by stephenlb** [comment]
All fixes added to the PR.

**Pull request merged**